torch/csrc/distributed/c10d/init.cpp

```diff
@@ -2670,6 +2671,13 @@ such as `dist.all_reduce(tensor, async_op=True)`.
         return ivalue.toCustomClass<::c10d::Work>();
       });
 
+  auto fakeProcessGroup =
```

Comment:
wow is this all the binding work needed?

do you need to add some .pyi stuff and docs?

Reply:
> wow is this all the binding work needed?

Yup. `torch::CustomClassHolder` is doing the heavy lifting here :)

> do you need to add some .pyi stuff and docs?

Nope. It is seen as `torch.distributed.distributed_c10d.ProcessGroup` in Python which is already covered.

```python
>>> import torch.distributed as dist
>>> from torch.testing._internal.distributed.fake_pg import FakeStore
>>> store = FakeStore()
>>> dist.init_process_group("fake", rank=0, world_size=2, store=store)
>>> print(dist.group.WORLD)
<torch.distributed.distributed_c10d.ProcessGroup object at 0x7f6a0df9ff70>
```